src/types/string.rs

```diff
@@ -136,24 +127,9 @@ impl<'a> IntoPy<PyObject> for &'a String {
 
 /// Allows extracting strings from Python objects.
 /// Accepts Python `str` and `unicode` objects.
-impl<'source> crate::FromPyObject<'source> for Cow<'source, str> {
+impl<'source> crate::FromPyObject<'source> for &'source str {
     fn extract(ob: &'source PyAny) -> PyResult<Self> {
-        <PyString as PyTryFrom>::try_from(ob)?.to_string()
-    }
-}
-
-/// Allows extracting strings from Python objects.
-/// Accepts Python `str` and `unicode` objects.
-impl<'a> crate::FromPyObject<'a> for &'a str {
-    fn extract(ob: &'a PyAny) -> PyResult<Self> {
-        let s: Cow<'a, str> = crate::FromPyObject::extract(ob)?;
-        match s {
-            Cow::Borrowed(r) => Ok(r),
-            Cow::Owned(r) => {
-                let r = ob.py().register_any(r);
```

Comment:
This was actually the only use of the api `register_any`. IMO it adds complexity and now doesn't have a clear use case, should we remove this API (or mark deprecated)?

We could discuss this in a separate issue.

Reply:
Good catch :+1:

Reply:
See #1023